### PR TITLE
Bug 884681 - Use custom verify/logout callbacks with SSO to handle remov...

### PR DESCRIPTION
**Pull request by mjschranz**
...al of username from session

https://bugzilla.mozilla.org/show_bug.cgi?id=884681

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -106,6 +106,39 @@ module.exports = function ( app, rawUrl ) {
...
+    personaLogout: function( error, req, res ) {
+      req.session.username = null;
+
+      if ( error ) {
+        return res.json( { status: "failure", reason: error } );
+      }
+      
+      res.json( { status: "okay" } );
     }
   }; // END LoginAPI
```
Comment (Pomax) [inline index.js]: any sensible editor will be able to signal this, no need a comment here.

File: index.js [hunk 2]
```diff
@@ -106,6 +106,39 @@ module.exports = function ( app, rawUrl ) {
...
+        req.session.username = webmaker.username;
+        return res.json( 200, {
+          status: "okay",
+          user: webmaker,
+          email: email
+        });            
+      });
+    },
+    personaLogout: function( error, req, res ) {
+      req.session.username = null;
```
Comment (Pomax) [inline index.js]: this should probably just be `delete req.session.username`

File: index.js [hunk 3]
```diff
@@ -107,38 +111,43 @@ module.exports = function ( app, rawUrl ) {
...
+        // Set session
+        req.session.username = webmaker.username;
+        return res.json( 200, {
+          status: "okay",
+          user: webmaker,
+          email: email
         });
+      });
+    },
+    logoutResponse: function( error, req, res ) {
```
Comment (jbuck) [inline index.js:158]: logoutResponse doesn't take an error parameter

File: package.json [hunk 4]
```diff
@@ -19,6 +19,7 @@
     "url": "https://github.com/mozilla/node-webmaker-loginapi.git"
   },
   "dependencies": {
+    "express-persona": "0.0.7",
```
Comment (jbuck) [inline package.json]: 0.0.9 is available on npm

File: index.js [hunk 5]
```diff
@@ -107,38 +111,43 @@ module.exports = function ( app, rawUrl ) {
...
 
-    loginAPI.getUser( personaUser, function( err, webmaker ) {
-      // Error handling
-      if ( err || !webmaker ) {
-        return res.json( 404, {
-          status: "failed",
-          reason: ( err || "webmaker not found" )
+      loginAPI.getUser( email, function( err, webmaker ) {
+        // Error handling
+        if ( err || !webmaker ) {
```
Comment (jbuck) [inline index.js]: This seems odd to me; how do you tell the difference between something like "I can't connect to the login server" and "This user doesn't have a webmaker account yet?"

File: index.js (hunk 3 above)
Comment (mjschranz) [inline index.js:158]: https://github.com/jbuck/express-persona/blob/master/index.js#L21-L29 seems to indicate otherwise.

**Pull request merged**